Ejercicio 1

P3rzeval opened the pull request.

P3rzeval committed:
Ejercicio 1

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+*/
+document.querySelector("#boton").onclick = function() {
+
+	const $numerosFamiliares = Number(document.querySelector("#numeros-familiares").value);
```
creo que estás usando demasiados espacios para indentar

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+
+	const $numerosFamiliares = Number(document.querySelector("#numeros-familiares").value);
+	
+	const $textoH2 = document.querySelector('h2');
```
`$titulo`

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+
+Punto bonus: Crear un botón para "empezar de nuevo" que empiece el proceso nuevamente, borrando los inputs ya creados (investigar cómo en MDN).
+*/
+document.querySelector("#boton").onclick = function() {
```
qué hace `#boton`? ponele un mejor nombre

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+	$textoH2.textContent = ('Ingrese la edad de sus intregrantes');
+	$botonCalcular.style.display = 'inline';
+		
+	function crearLabelInput ($numerosFamiliares) {
+		
+		for(i =0; i < $numerosFamiliares; i++) {
+
+			const nodoPagina = document.querySelector('#section');
+			const nuevoInput = document.createElement('input');
+			const nuevolabel = document.createElement('label');
+			const textoLabel = document.createTextNode('Edad');
+
+			nodoPagina.appendChild(nuevolabel);
+			nodoPagina.appendChild(nuevoInput);
+			nuevolabel.appendChild(textoLabel);
+			nuevoInput.setAttribute("class", "edades")
+			nuevoInput.type = "number";
+		}
+	};
```
esta función uría por fuera, tampoco le podés poner crearLabelInput porque lo que hace en todo caso es "crearIntegrante" o algo por el estilo.

También ojo con mezclar `'` y `"`

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+
+	resultados.style.display = "inline-block";
+
+	function calcularEdadMayor(edades) {
+
+		let edadMayor = [];
+
+		for(let i =0; i < edades.length; i++){
+			if(edadMayor < Number(edades[i].value)){
+				edadMayor[0] = Number(edades[i].value);
+			}
+		}
+
+		document.querySelector('#edad-mayor').innerText = edadMayor;
+
+	};
```
esto va por fuera, y edades tiene que ser un listado de números, no de inputs.

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+
+	};
+
+	function calcularEdadMenor(edades) {
+
+		let edadMenor = Number(edades[0].value);
+
+		for(let i =0; i < edades.length; i++){
+
+			let numeros = Number(edades[i].value);
+
+			if( numeros < edadMenor ){
+				edadMenor = numeros;
+			}
+		}
+		document.querySelector('#edad-menor').textContent = edadMenor;
+	};
+	
+	function calcularPromedioEdad(edades) {
+
+		let contador = 0;
+		for(let i =0; i < edades.length; i++) {
+			contador += Number(edades[i].value);
+		}
+		const edadPromedio = contador / edades.length;
+		
+		document.querySelector('#edad-promedio').textContent =  edadPromedio;
+	}
```
todas estas son historias globales

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+		}
+		const edadPromedio = contador / edades.length;
+		
+		document.querySelector('#edad-promedio').textContent =  edadPromedio;
```
esta función no tiene que saber de actualizar elementos del HTML, sólo tiene que saber de devolver el promedio, lo mismo con las otras.

fsodano commented on tareas/clase-6/tarea-1-clase-6.js:
```diff
@@ -0,0 +1,103 @@
+		document.querySelector('#edad-promedio').textContent =  edadPromedio;
+	}
+
+	calcularEdadMayor(edades);
+	calcularEdadMenor(edades);
+	calcularPromedioEdad(edades);
```
acá es donde agarrarías el resultado de los cálculos y los pondrías en el HTML

P3rzeval committed:
Correcciones

se corrigio la indentación y  las funciones se volvieron globales y se renombraron algunas variables para mejor sentido logico.
